src/DataStructures/Python/ToNumpy.hpp

```diff
@@ -4,12 +4,44 @@
 #pragma once
 
 #include <Python.h>
+#include <array>
```

Comment:
I think you should `include` `Matrix.hpp` instead of forward declaring since you actually use the `Matrix` class in this header file.